### ISSUE TITLE
Remove all the warnings generated by Sphinx

### ISSUE THREAD

**Issue by agarny** [cellml/cellml-specification#134]
`make html` and `make latexpdf` result in a lot of warnings being generated by Sphinx. There is no justification for them so they ought to be addressed. For instance, for `make html` (using https://github.com/hsorby/cellml-specification/tree/local_build):

```
sphinx-build -b html -d build/doctrees   src build/html
Running Sphinx v2.4.4
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 132 source files that are out of date
updating environment: [new config] 132 added, 0 changed, 0 removed
reading sources... [100%] reference/sectionD_references
src/reference/sectionA_definitions.inc:6: WARNING: duplicate label speca_terminology, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specA1.rst
src/reference/sectionA_definitions.inc:6: WARNING: duplicate label speca_cellml_information_sets, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specA2.rst
src/reference/sectionA_definitions.inc:43: WARNING: duplicate label speca_semantic_equivalence, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specA2.rst
src/reference/sectionA_definitions.inc:6: WARNING: duplicate label speca_data_representation_formats, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specA3.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label model, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB01.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label import, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB02.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label import_units, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB03.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label import_component, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB04.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label units, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB05.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label unit, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB06.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label component, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB07.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label variable, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB08.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label reset, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB09.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label test_value, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB10.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label reset_value, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB11.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label math, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB12.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label encapsulation, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB13.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label component_ref, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB14.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label connection, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB15.rst
src/reference/sectionB_elements.inc:6: WARNING: duplicate label map_variables, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB16.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_interpretation_of_imports, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC01_interpretation_of_imports.rst
src/reference/sectionC_interpretation.inc:6: WARNING: duplicate label specc_units_reference, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC02_units_reference.rst
src/reference/sectionC_interpretation.inc:6: WARNING: duplicate label specc_interpretation_of_units, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC03_interpretation_of_units.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_component_reference, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC04_component_reference.rst
src/reference/sectionC_interpretation.inc:6: WARNING: duplicate label specc_variable_reference, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC05_variable_reference.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_interpretation_of_initial_values, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC06_interpretation_of_initial_values.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_effect_of_units_on_variables, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC07_effect_of_units_on_variables.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_interpretation_of_mathematics, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC08_interpretation_of_mathematics.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_interpretation_of_encapsulation, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC09_interpretation_of_encapsulation.rst
src/reference/sectionC_interpretation.inc:6: WARNING: duplicate label specc_interpretation_of_map_variables, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC10_interpretation_of_map_variables.rst
src/reference/sectionC_interpretation.inc:7: WARNING: duplicate label specc_interpretation_of_variable_resets, other instance in /Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC11_interpretation_of_variable_resets.rst
looking for now-outdated files... /Users/Alan/Desktop/cellml-specification/src/formal.rst:46: WARNING: reference/sectionD_references is already assigned section numbers (nested numbered toctree?)
none found
pickling environment... done
checking consistency... /Users/Alan/Desktop/cellml-specification/src/formal.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] reference/sectionD_references
/Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB07.rst:48: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specB12.rst:28: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/Users/Alan/Desktop/cellml-specification/src/reference/formal_and_informative/specC03_interpretation_of_units.rst:46: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/Users/Alan/Desktop/cellml-specification/src/reference/informative/informB12_math4.rst:28: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/Users/Alan/Desktop/cellml-specification/src/reference/informative/informB7_component3.rst:48: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/Users/Alan/Desktop/cellml-specification/src/reference/informative/informC03_interpretation_of_units1.rst:46: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
generating indices...  genindexdone
writing additional pages...  search/Users/Alan/Desktop/venv_specification/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:21: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {% endblock %}
done
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 39 warnings.

The HTML pages are in build/html.

Build finished. The HTML pages are in build/html.
```

**Comment by hsorby** [comment]
I think I have a handle on why this is happening and will address this issue once the local build PR is merged.

**Comment by kerimoyle** [comment]
It's because of sections being included into more than one place.  I think when this is split into normative/informative it might solve itself?  I've tried it in variety of different arrangements (labels inside/outside the marker tags; inside/outside the files etc) and went with the one that gave the functionality I wanted (though clearly still throws errors).

**Comment by MichaelClerx** [comment]
That seems to be most of them yes!

There's also these at the end though:
```
/home/michael/dev/cellml/specification/src/reference/formal_and_informative/specB07.rst:48: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/home/michael/dev/cellml/specification/src/reference/formal_and_informative/specB12.rst:28: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/home/michael/dev/cellml/specification/src/reference/formal_and_informative/specC03_interpretation_of_units.rst:46: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/home/michael/dev/cellml/specification/src/reference/informative/informB12_math4.rst:28: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/home/michael/dev/cellml/specification/src/reference/informative/informB7_component3.rst:48: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
/home/michael/dev/cellml/specification/src/reference/informative/informC03_interpretation_of_units1.rst:46: WARNING: Could not lex literal_block as "xml". Highlighting skipped.
```

**Comment by kerimoyle** [comment]
The lexing is fixed in one of mine.

**Comment by kerimoyle** [comment]
See 2f392d3 and 7450a46

**Comment by hsorby** [comment]
I have fixed all the warnings that I have on macOS is anyone else still getting warnings?

**Comment by agarny** [comment]
I have just tried using the `cellml-2-drafting` branch:

```
make html
sphinx-build -b html -d build/doctrees   src build/html
Running Sphinx v2.4.4
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 99 source files that are out of date
updating environment: [new config] 99 added, 0 changed, 0 removed
reading sources... [100%] reference/libcellml/libcellmlC03_interpretation_of_units
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB5.rst:12: WARNING: Unknown interpreted text role "libcellml_tutorials".
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB6.rst:12: WARNING: Unknown interpreted text role "libcellml_tutorials".
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informB9_reset1.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC01_imports.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC02_units_reference.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC04_component_reference.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC05_variable_reference.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC07_effect_of_units_on_variables.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC08_interpretation_of_mathematics.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC09_interpretation_of_encapsulation.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC10_interpretation_of_map_variables.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/informative/informC11_interpretation_of_variable_resets.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB1.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB12_math.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB15_connection.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB16_map_variables.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB2.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB3.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB4.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB5.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB6.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB7.rst: WARNING: document isn't included in any toctree
/Users/Alan/Work/Programming/cellml-specification/src/reference/libcellml/libcellmlB8.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] reference/libcellml/libcellmlC03_interpretation_of_units
generating indices...  genindexdone
writing additional pages...  search/Users/Alan/cellml-specification-virtualenv/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:21: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {% endblock %}
done
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 23 warnings.

The HTML pages are in build/html.

Build finished. The HTML pages are in build/html.
```

but I seem to remember that @kerimoyle thinks it's "normal"?

**Comment by hsorby** [comment]
that is odd

**Comment by hsorby** [comment]
Ahh this is building the combined documentation. I must go back and check mine.

**Comment by kerimoyle** [comment]
Haha, it's not so much "normal" as "understood-and-in-process-of-fixing".  There are placeholder files for the informative spec, whose inclusion I've commented out so that we don't see **TODO** statements everywhere.  As soon as they're written, then they will be re-included and the warnings will disappear.

**Comment by agarny** [comment]
Ok, I thought I would try to build the spec from scratch following exactly the instructions at https://github.com/cellml/cellml-specification. Basically, the following commands worked as expected:
```
git clone https://github.com/cellml/cellml-specification.git
virtualenv venv_specification
source venv_specification/bin/activate
pip install sphinx sphinx_rtd_theme
pip install -r cellml-specification/requirements.txt
cd cellml-specification
```
However, `make html`, which comes next, didn't. I am now getting:
```
sphinx-build -b html -d build/doctrees   src build/html
Running Sphinx v3.0.0
making output directory... done

Exception occurred:
  File "/Users/Alan/Desktop/venv_specification/lib/python3.7/site-packages/hoverxref/extension.py", line 198, in setup_sphinx_tabs
    for listener_id, function in app.events.listeners.get('html-page-context').items():
AttributeError: 'list' object has no attribute 'items'
The full traceback has been saved in /var/folders/k8/gmbwgb910z52mrztjsdwjpbh0000gn/T/sphinx-err-r168fsz_.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```
@hsorby, any idea of what might be going on here?

**Comment by hsorby** [comment]
I would start guessing that something has changed between Sphinx 2.4 and 3.0.  I'm guessing you recently upgraded brew packages?

**Comment by agarny** [comment]
No, my system version of Sphinx is still 2.4.4:
```
$ which sphinx-build
/usr/local/opt/sphinx-doc/bin/sphinx-build
$ sphinx-build --version
sphinx-build 2.4.4
```
It would seem that Homebrew hasn't updated Sphinx yet. However, within virtualenv, I have a different version of Sphinx:

```
$ source virtualenv/cellml-specification/bin/activate
(cellml-specification) $ which sphinx-build
/Users/Alan/virtualenv/cellml-specification/bin/sphinx-build
(cellml-specification) $ sphinx-build --version
sphinx-build 3.0.0
```

Still, it looks like there might indeed be a (big) difference between Sphinx 2.4.4 and 3.0.0. Yeah!

**Comment by MichaelClerx** [comment]
Do we need this as a separate issue? I imagine we won't declare the informative spec "finished" until all the warnings are gone?

**Comment by MichaelClerx** [comment]
@agarny can we close this ticket now as a combination of:

- informative/libcellml isn't ready yet, so normative should be default #167 
- remaining errors are from #281 

?

**Comment by agarny** [comment]
With `make html` on `cellml-2-drafting`, I am only getting one message now:
```
writing additional pages...  search/Users/Alan/virtualenv/cellml-specification/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:21: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
```
Not quite sure why since I am now using Sphinx 2.4.0. As for `make latexpdf`, I am only getting loads of warnings from latexpdf itself, so I guess we can close this issue.
